crates/state/src/error.rs

```diff
@@ -0,0 +1,19 @@
+use std::fmt::Debug;
+
+pub type Result<T> = std::result::Result<T, GlobalStateError>;
+
+#[derive(Debug, thiserror::Error)]
+pub enum GlobalStateError {
+    #[error("Invalid item.")]
+    InvalidItem,
```

Comment:
Can you please put one line between each item (I find it a bit distraction due to the proc-macros used)